### PR TITLE
Feature | Thoughts

### LINKED ISSUE #249: Bounty 6$ | Feature | Thought
### What ?
- At the moment, users can write blogs. Blogs are long form content. 
- Introducing Thoughts. User can write a thought. Thoughts are short form. Think of 1 Thought like a Twitter post or Threads post. 

### How ?

1. backend/prisma/schema.prisma - 

```javascript
model Thought {
  id            String     @id @default(uuid())
  content       String
  publishedDate DateTime   @default(now())
  published     Boolean    @default(false)
  author        User       @relation(fields: [authorId], references: [id], onDelete: Cascade)
  authorId      String
}
```
Run prisma migrations to update db with schema

2. backend/src/routes/thought.ts - thoughtRouter
```bash
thoughtRouter.post
thoughtRouter.put
thoughtRouter.get: 1 and all
thoughtRouter.delete
```
create, update, delete - protected routes
get - public routes

3. Test all routes on Postman (optional)

<hr />

4. frontend/src/pages/Thought.ts - 
```bash
1. Create Thought button 
2. Show all Thoughts
3. Delete Thought 
4. Update Thought
```

The frontend should be simple, page/Thought.ts is a temporary placeholder to test the backend setup. We will eventually integrate it into the App.

### PR DESCRIPTION
**Pull request by uanik03**
## Description
At the moment, users can write blogs. Blogs are long form content.  User can write a thought. Thoughts are short form.

Resolves #249 

## Linked Issues
#249 

## Type of Change
 

- New feature (non-breaking change which adds functionality)

## Changes

- backend/src/routes/thought.ts - thoughtRouter (CRUD ops for thought feature)
- frontend/src/pages/Thought.ts - (page to test all features)
- ...

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I assure there are no similar/duplicate pull requests regarding the same issue
- [x] My changes follow the project's coding guidelines and best practices
- [x] Code is formatted properly and lint check passes successfully
- [x] I have made corresponding changes to the documentation (if applicable)

### CODE REVIEW

**Comment by uanik03** [comment]
change the BACKEND_URL  in config.ts

File: frontend/src/config.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-export const BACKEND_URL = 'https://backend.aadesh-kulkarni08.workers.dev';
+// export const BACKEND_URL = 'https://backend.aadesh-kulkarni08.workers.dev';
+export const BACKEND_URL = 'http://localhost:39013';
```
Comment (aadeshkulkarni) [inline frontend/src/config.ts:2]: The BACKEND_URL should be pointing to workers.dev, not localhost. Please rollback changes to this file.

File: frontend/src/components/Appbar.tsx [hunk 2]
```diff
@@ -72,6 +72,14 @@ const Appbar = ({ skipAuthCheck = false, pageActions, hideWriteAction = false }:
...
                 </Link>
               )}
+              <Link to="/thoughts">
+              <button
+                    type="button"
+                    className="focus:outline-none hover:bg-sub rounded-3xl focus:ring-4 focus:ring-gray-100 font-medium flex items-center gap-2 text-sm px-3 md:px-5 py-2.5"
+                  >
+                    <WriteIcon /> thought
+                  </button>
+              </Link>
```
Comment (aadeshkulkarni) [inline frontend/src/components/Appbar.tsx:82]: Fix indentation

File: backend/prisma/migrations/20240713163709_thought/migration.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+-- CreateTable
+CREATE TABLE "Thought" (
+    "id" TEXT NOT NULL,
+    "content" TEXT NOT NULL,
+    "publishedDate" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "published" BOOLEAN NOT NULL DEFAULT false,
+    "authorId" TEXT NOT NULL,
```
Comment (aadeshkulkarni) [inline backend/prisma/migrations/20240713163709_thought/migration.sql:7]:
```javascript
 author        User       @relation(fields: [authorId], references: [id], onDelete: Cascade)
 authorId      String
 ```

File: backend/prisma/migrations/20240713163709_thought/migration.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+-- CreateTable
+CREATE TABLE "Thought" (
+    "id" TEXT NOT NULL,
```
Comment (aadeshkulkarni) [inline backend/prisma/migrations/20240713163709_thought/migration.sql:3]:
```javascript
id            String     @id @default(uuid())
```

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -30,6 +31,8 @@ function App() {
               <Route path="/signin" element={<Signin />} />
               <Route path="/blogs" element={<Blogs />} />
               <Route path="/blog/:id" element={<Blog />} />
+              <Route path="/thoughts" element={<Thought />} />
+
```
Comment (aadeshkulkarni) [inline frontend/src/App.tsx:35]: Remove extra line

**Review by aadeshkulkarni — CHANGES_REQUESTED** [review]

File: backend/prisma/migrations/20240713163709_thought/migration.sql (hunk 3 above)
Comment (uanik03) [reply backend/prisma/migrations/20240713163709_thought/migration.sql:7]: @aadeshkulkarni what should I change here?

**Pull request closed without merging**